### ISSUE TITLE
All Good Except For One

### ISSUE THREAD

**Issue by johnrs** [xlab/handysort#5]
It looks like end-of-string is sorting after numbers, rather than before.
[]string{"", "1", "a", "a1"]  ==>  [1  a1 a]

**Comment by xlab** [comment]
:stuck_out_tongue_winking_eye::gun:

**Comment by xlab** [comment]
Consider using https://github.com/fvbommel/util/tree/master/sortorder though :)

**Comment by johnrs** [comment]
I’m sorry to have so much bad news.  J

John

```
John Souvestre - New Orleans LA
```

**Comment by johnrs** [comment]
Unfortunately, something else is wrong now.

[]string{"#", "1"} è [# 1]

Everything else looks good.  Including {"a#", "a1"} which sorts correctly as [a1 a#].

John

```
John Souvestre - New Orleans LA
```

**Comment by johnrs** [comment]
Yes, it is nice.  I have given him a few minor improvements.  I’m working to add handling all of the Unicode 0 digits to it now, but it slows it down.

John

```
John Souvestre - New Orleans LA
```

**Comment by xlab** [comment]
I wonder why OS X does that
![sort](https://www.dropbox.com/s/y3ikkfpo00zwjlz/Screenshot%202015-04-22%2016.20.14.png?dl=1)

So I'm expecting `#` < `1` but then it's not clear why `a1` >  `a#`.

**Comment by johnrs** [comment]
Probably because it’s easy to do, since that is how the characters are ordered.  But in a natural sort the idea is that “numbers are different”.  So it doesn’t make much sense that a number would sort between two non-number characters.

A better visual example would be a “space” character instead of the “#”.  The way it sorts  now:  [  1 _].  The number is sorting between two non-numbers.  What I think it should be:  [1   _].

Imagine filenames, some with space separators and others with underscore separators.  Shouldn’t they sort adjacently, without numbers between them?

Given this, should numbers sort before or after non-numbers?  Either way seems OK to me, but every implementation I’ve looked at has them first.

John

```
John Souvestre - New Orleans LA
```
